services/datamanager/builtin/sync/upload_arbitrary_file.go

```diff
@@ -0,0 +1,156 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/pkg/errors"
+	v1 "go.viam.com/api/app/datasync/v1"
+
+	"go.viam.com/rdk/logging"
+)
+
+// UploadChunkSize defines the size of the data included in each message of a FileUpload stream.
+var (
+	UploadChunkSize            = 64 * 1024
+	errFileEmpty               = errors.New("file is empty (0 bytes)")
+	errFileModifiedTooRecently = errors.New("file modified too recently")
+)
+
+func uploadArbitraryFile(
```

Comment:
either here or someplace else, it'd be good to add a comment clarifying what "arbitrary files" mean in this context -- files not created by our capture service. this phrase keeps tripping me personally :)

Reply:
added